File: Frontend/Mobile/lib/models/community/post.dart
class Post {
  final int id;
  final String content;
  final String createdAt;
  final Child? child;
  final List<Media> media;
  final int commentsCount;

  Post({
    required this.id,
    required this.content,
    required this.createdAt,
    this.child,
    required this.media,
    required this.commentsCount,
  });

  factory Post.fromJson(Map<String, dynamic> json) {
    return Post(
      id: json['id'],
      content: json['content'] ?? '',
      createdAt: json['createdAt'] ?? '',
      child: json['child'] != null
          ? Child.fromJson(json['child'])
          : null,
      media: json['media'] != null
          ? (json['media'] as List)
          .map((e) => Media.fromJson(e))
          .toList()
          : [],
      commentsCount: json['commentsCount'] ?? 0,
    );
  }
}

class Child {
  final int id;
  final String name;

  Child({
    required this.id,
    required this.name,
  });

  factory Child.fromJson(Map<String, dynamic> json) {
    return Child(
      id: json['id'],
      name: json['name'] ?? '',
    );
  }
}

class Media {
  final int id;
  final String type;
  final String url;

  Media({
    required this.id,
    required this.type,
    required this.url,
  });

  factory Media.fromJson(Map<String, dynamic> json) {
    return Media(
      id: json['id'],
      type: json['type'] ?? '',
      url: json['url'] ?? '',
    );
  }
}

File: Backend/src/main/java/org/example/backend/model/community/Post.java
package org.example.backend.model.community;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.example.backend.model.Child;
import org.example.backend.model.Kit;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "posts")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Post {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Column(columnDefinition = "TEXT")
    private String content;

    private LocalDateTime createdAt;

    @ManyToOne
    @JoinColumn(name = "child_id")
    private Child child;

    @ManyToOne
    @JoinColumn(name = "kit_id")
    private Kit kit;

    @OneToMany(mappedBy = "post", cascade = CascadeType.ALL)
    private List<Comment> comments;

    @OneToMany(mappedBy = "post", cascade = CascadeType.ALL)
    private List<PostLike> likes;

    @OneToMany(mappedBy = "post", cascade = CascadeType.ALL)
    private List<Media> media;
}
